bundles/org.eclipse.jface.text/src/org/eclipse/jface/contentassist/CompletionProposalDrawSupport.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2024 SAP SE.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     SAP SE - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jface.contentassist;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.Listener;
+import org.eclipse.swt.widgets.Table;
+
+import org.eclipse.jface.internal.text.TableOwnerDrawSupport;
+
+/**
+ * Provides custom drawing support for completion proposals. This class ensures that completion
+ * proposals are always rendered with a focused appearance.
+ *
+ * <p>
+ * This drawing behavior addresses the particular situation where the code completion is triggered
+ * via keyboard shortcut, leaving the editor focused. In such cases, without this custom drawing
+ * support, the completion proposal would appear unfocused, leading to a suboptimal coloring.
+ * </p>
+ */
+public class CompletionProposalDrawSupport implements Listener {
+
+	private TableOwnerDrawSupport fTableOwnerDrawSupport;
+
+	private CompletionProposalDrawSupport(Table table) {
+		fTableOwnerDrawSupport= new TableOwnerDrawSupport(table);
+	}
+
+	public static void install(Table table) {
+		CompletionProposalDrawSupport listener= new CompletionProposalDrawSupport(table);
+		table.addListener(SWT.Dispose, listener);
+		table.addListener(SWT.MeasureItem, listener);
+		table.addListener(SWT.EraseItem, listener);
+		table.addListener(SWT.PaintItem, listener);
+	}
+
+	@Override
+	public void handleEvent(Event event) {
+		Color background= event.item.getDisplay().getSystemColor(SWT.COLOR_TITLE_BACKGROUND);
+		Color forground= event.item.getDisplay().getSystemColor(SWT.COLOR_WHITE);
```

Comment:
is setting "white" correct for all themes?

Reply:
Not sure, at least on MacOS I'm not able to change the selection foreground color. It is always white.
I think there is no system color for selection foreground, or any ideas which one I could use?

Reply:
On my Mac, SWT.COLOR_WIDGET_HIGHLIGHT_SHADOW and SWT.COLOR_LIST_BACKGROUND does result in a white text color. But both don't seem to be right here.